### PR TITLE
[Scrutinizer] Fix issues in mod/ (_well_known to contactgroup)

### PR DESCRIPTION
**Pull request by MrPetovan**
Follow-up to #4166

### CODE REVIEW

File: include/datetime.php [hunk 1]
```diff
@@ -535,16 +536,14 @@ function cal($y = 0,$m = 0, $links = false, $class='') {
 	$o .= '</tr><tr>';
 
 	while ($d <= $l) {
-		if (($dow == $f) && (! $started)) {
-			$started = true;
-		}
+		$started = (($dow == $f) && (!$started));
```
Comment (annando) [inline include/datetime.php]: This is a difference in functionality. The original code would never set ```$started``` back to "false". This code is able to do so. Is it intended?

File: mod/admin.php [hunk 2]
```diff
@@ -720,7 +709,7 @@ function admin_page_summary(App $a)
 	if (Config::get('system', 'check_new_version_url', 'none') != 'none') {
 		$gitversion = Config::get('system', 'git_friendica_version');
 		if (version_compare(FRIENDICA_VERSION, $gitversion) < 0) {
-			$warningtext[] = sprintf(t('There is a new version of Friendica available for download. Your current version is %1$s, upstream version is %2$s'), $FRIENDICA_VERSION, $gitversion);
+			$warningtext[] = sprintf(t('There is a new version of Friendica available for download. Your current version is %1$s, upstream version is %2$s'), FRIENDICA_VERSION, $gitversion);
```
Comment (annando) [inline mod/admin.php]: ```t()``` includes ```sprintf()``` doesn't it?

File: mod/admin.php [hunk 3]
```diff
@@ -2123,12 +2092,10 @@ function __get_theme_admin_form(App $a, $theme)
 	// reload active themes
 	if (x($_GET, "a") && $_GET['a'] == "r") {
 		check_form_security_token_redirectOnErr(System::baseUrl() . '/admin/themes', 'admin_themes', 't');
-		if ($themes) {
-			foreach ($themes as $th) {
-				if ($th['allowed']) {
-					uninstall_theme($th['name']);
-					install_theme($th['name']);
-				}
+		foreach ($themes as $th) {
```
Comment (annando) [inline mod/admin.php]: I haven't checked: Can we be sure that ```$themes``` is always an array?

File: mod/cal.php [hunk 4]
```diff
@@ -72,146 +76,142 @@ function cal_init(App $a) {
...
 			intval($a->profile['profile_uid'])
 		);
 		if (DBM::is_result($r)) {
-			$contact = $r[0];
 			$remote_contact = true;
 		}
 	}
-	if(! $remote_contact) {
-		if(local_user()) {
-			$contact_id = $_SESSION['cid'];
```
Comment (annando) [inline mod/cal.php]: Is this variable unused?

File: include/datetime.php (hunk 1 above)
Comment (MrPetovan) [reply include/datetime.php]: No, good catch, I went too fast on my formatting spree.

File: mod/admin.php (hunk 2 above)
Comment (MrPetovan) [reply mod/admin.php]: Yep, I'll simplify the call. I'm not concentrating on it because it isn't a code error per se.

File: mod/admin.php (hunk 3 above)
Comment (MrPetovan) [reply mod/admin.php]: Defined as such on line 1963, populated on line 1982.

File: mod/cal.php (hunk 4 above)
Comment (MrPetovan) [reply mod/cal.php]: After line 129, no. If you think this call should be moved up let me know.

File: mod/admin.php [hunk 5]
```diff
@@ -1266,7 +1266,7 @@ function admin_page_site(App $a)
 		'$banner'		=> array('banner', t("Banner/Logo"), $banner, ""),
 		'$shortcut_icon'	=> array('shortcut_icon', t("Shortcut icon"), Config::get('system','shortcut_icon'),  t("Link to an icon that will be used for browsers.")),
 		'$touch_icon'		=> array('touch_icon', t("Touch icon"), Config::get('system','touch_icon'),  t("Link to an icon that will be used for tablets and mobiles.")),
-		'$info'			=> array('info', t('Additional Info'), $info, sprintf(t('For public servers: you can add additional information here that will be listed at %s/servers.'), get_server())),
+		'$info'			=> array('info', t('Additional Info'), $info, t('For public servers: you can add additional information here that will be listed at %s/servers.', System::baseUrl()), get_server()),
```
Comment (annando) [inline mod/admin.php]: Why had you added ```System::baseUrl())``` here?

File: mod/admin.php [hunk 6]
```diff
@@ -2121,7 +2123,7 @@ function admin_page_themes(App $a)
 		'$function'            => 'themes',
 		'$plugins'             => $plugins,
 		'$pcount'              => count($themes),
-		'$noplugshint'         => sprintf(t('No themes found on the system. They should be paced in %1$s'),'<code>/view/themes</code>'),
+		'$noplugshint'         => t('No themes found on the system. They should be paced in %1$s', '<code>/view/themes</code>'),
```
Comment (annando) [inline mod/admin.php]: ```placed```

File: mod/admin.php [hunk 7]
```diff
@@ -1441,11 +1441,11 @@ function admin_page_dbsync(App $a)
  */
 function admin_page_users_post(App $a)
 {
-	$pending     = (x($_POST, 'pending')           ? $_POST['pending']           : array());
-	$users       = (x($_POST, 'user')              ? $_POST['user']		      : array());
-	$nu_name     = (x($_POST, 'new_user_name')     ? $_POST['new_user_name']     : '');
-	$nu_nickname = (x($_POST, 'new_user_nickname') ? $_POST['new_user_nickname'] : '');
-	$nu_email    = (x($_POST, 'new_user_email')    ? $_POST['new_user_email']    : '');
+	$pending     = defaults($_POST, 'pending'          , array());
```
Comment (annando) [inline mod/admin.php]: Just a style question: Wouldn't it be preferable to set the comma directly after the parameter and not with much space before?

File: mod/cal.php (hunk 4 above)
Comment (annando) [reply mod/cal.php]: I meant the assignment of the variable with the content of ```$_SESSION['cid'];``` which now doesn't happen in the code any more if I see it right.

File: mod/admin.php (hunk 5 above)
Comment (MrPetovan) [reply mod/admin.php]: If you look at the translation string, there is a `%s` but no interpolation variable was provided. I just filled in the blank.

File: mod/admin.php (hunk 7 above)
Comment (MrPetovan) [reply mod/admin.php]: To align with the previous `defaults()` calls default values.

File: mod/admin.php (hunk 5 above)
Comment (annando) [reply mod/admin.php]: The ```%s``` belongs to ```get_server()```.

Comment (MrPetovan) [reply mod/admin.php]: 😅

File: mod/common.php [hunk 8]
```diff
@@ -45,37 +46,36 @@ function common_content(App $a) {
...
-	if(! $cid) {
-		if(get_my_url()) {
-			$r = q("SELECT `id` FROM `contact` WHERE `nurl` = '%s' AND `uid` = %d LIMIT 1",
-				dbesc(normalise_link(get_my_url())),
-				intval($profile_uid)
+	if (!$cid && get_my_url()) {
+		/// @todo : Initialize $profile_uid
+		$r = q("SELECT `id` FROM `contact` WHERE `nurl` = '%s' AND `uid` = %d LIMIT 1",
+			dbesc(normalise_link(get_my_url())),
+			intval($profile_uid)
```
Comment (MrPetovan) [inline mod/common.php]: @annando Any idea how to properly initialize `$profile_uid` here?

Comment (annando) [reply mod/common.php]: Isn't this the local_user() in most cases?

Comment (MrPetovan) [reply mod/common.php]: Maybe, but this is the module showing the common friends between two users/contacts, so there are many possibilities.

Comment (annando) [reply mod/common.php]: Common friends is done differently. But for a deeper look I'm too tired now.

Comment (MrPetovan) [reply mod/common.php]: That's okay, it's been broken like this for a while, it can wait a little longer.

**Pull request merged**